src/System.Drawing.Common/src/System/Drawing/Font.cs

```diff
@@ -275,8 +275,55 @@ public override string ToString()
                                     _gdiCharSet,
                                     _gdiVerticalFont);
         }
-        
+
         // This is used by SystemFonts when constructing a system Font objects.
         internal void SetSystemFontName(string systemFontName) => _systemFontName = systemFontName;
+
+        public unsafe void ToLogFont(object logFont, Graphics graphics)
+        {
+            if (logFont == null)
+            {
+                throw new ArgumentNullException(nameof(logFont));
+            }
+
+            if (Marshal.SizeOf(logFont.GetType()) != sizeof(SafeNativeMethods.LOGFONT))
+            {
+                // If we don't actually have an object that is LOGFONT in size, trying to pass
+                // it to GDI+ is likely to cause an AV.
+                throw new ArgumentException();
+            }
+
+            SafeNativeMethods.LOGFONT nativeLogFont = ToLogFontInternal(graphics);
+            Marshal.PtrToStructure(new IntPtr(&nativeLogFont), logFont);
+        }
+
+        private unsafe SafeNativeMethods.LOGFONT ToLogFontInternal(Graphics graphics)
+        {
+            if (graphics == null)
+            {
+                throw new ArgumentNullException(nameof(graphics));
+            }
+
+            SafeNativeMethods.LOGFONT logFont = new SafeNativeMethods.LOGFONT();
+            Gdip.CheckStatus(Gdip.GdipGetLogFontW(
+                new HandleRef(this, NativeFont), new HandleRef(graphics, graphics.NativeGraphics), ref logFont));
+
+            // Prefix the string with '@' if this is a gdiVerticalFont.
+            if (_gdiVerticalFont)
+            {
+                int fullLength = logFont.lfFaceName.Length;
+                Span<char> temp = stackalloc char[fullLength];
+                temp[0] = '@';
+                logFont.lfFaceName.Slice(0, fullLength - 1).CopyTo(temp.Slice(1));
```

Comment:
If the original `lfFaceName` was `fullLength - 1` + null terminator, the modified `lfFaceName` is going to be missing the null terminator. Is it a problem? The docs say that `lfFaceName` should be null terminated.

Reply:
The buffer is fixed, so I don't think so? We certainly didn't check before. I'll double check.

Reply:
I don't think it really needs it, but the docs don't give the wiggle room. Checking my machine with > 1000 font results I don't have any that totally fill the buffer (i.e. all are null terminated). I'll force the null in. I suspect that this code has always been broken, I don't see any reference to the APIs allowing font names to be preceded with `@`. I suspect what we _should_ be doing is rolling the `@` back out of the returned `LOGFONT` and passing it back in. I'll create an issue to investigate this.